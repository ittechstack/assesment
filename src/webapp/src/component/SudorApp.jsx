import React, {useEffect, useState} from 'react';
import {TrainersTable} from "./TrainersTable";
import {ToastContainer} from "react-toastify";
import {WorkoutsTable} from "./WorkoutsTable";
import {getAllTrainers, getWorkoutsOfTrainer} from "../service/ApiService";

export const SudorApp = props => {

    const [trainers, setTrainers] = useState([]);
    const [selectedTrainer, setSelectedTrainer] = useState();

    const refreshTrainers = () => {
        getAllTrainers().then(resp => {
            setTrainers(resp.data);
        })
    };

    const refreshWorkoutsOfTrainer = (trainer) => {
        getWorkoutsOfTrainer(trainer).then(resp => {
            setSelectedTrainer({
                ...trainer,
                workouts: resp.data
            })
        })
    };

    useEffect(() => {
        refreshTrainers();
    }, []);

    return (
        <>
            <ToastContainer/>
            <h1>Sudor Application</h1>
            <div className="container">
                <div className="row">
                    <TrainersTable trainers={trainers} refreshWorkoutsOfTrainer={refreshWorkoutsOfTrainer}
                                   refreshTrainers={refreshTrainers} selectedTrainer={selectedTrainer}/>
                    <WorkoutsTable trainer={selectedTrainer} refreshWorkoutsOfTrainer={refreshWorkoutsOfTrainer}/>
                </div>
            </div>
        </>
    )
};