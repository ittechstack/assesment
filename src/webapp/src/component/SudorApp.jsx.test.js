import {SudorApp} from "./SudorApp";
import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

describe('SudorApp component testing', () => {
    it('render correctly SudorApp component', () => {
        const wrapper = renderer.create(<SudorApp/>).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render header', () => {
        const wrapper = shallow(<SudorApp/>);
        expect(wrapper.contains(<h1>Sudor Application</h1>)).toBe(true)
    });
});