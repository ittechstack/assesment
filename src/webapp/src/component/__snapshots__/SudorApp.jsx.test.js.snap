// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SudorApp component testing render correctly SudorApp component 1`] = `
Array [
  <div
    className="Toastify"
  />,
  <h1>
    Sudor Application
  </h1>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm"
      >
        <h3>
          Trainers
        </h3>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Trainer Name
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
        <div
          className="row"
        >
          <button
            className="btn btn-success"
            onClick={[Function]}
          >
            Add Trainer
          </button>
        </div>
      </div>
      <div
        className="col-sm"
      />
    </div>
  </div>,
]
`;
